### ISSUE TITLE
Add tags at counter level and allow instance tag have another name

### ISSUE THREAD

**Issue by zeugfr** [poxet/Influx-Capacitor#22]
After discussions with collegues, they argued that possibility to create many tags on InfluxDB can be an advantage to make some global queries more simple and get most advantage of InfluxDB features.
For example, creating "perfmon.processor.pct_user", "perfmon.processor.pct_system" and "perfmon.processor.pct_irq" will quickly force us to make many queries and merge the results.

Using only the field names of perfmon (countercategory, countername, instance) is too a little restrictive.

It should be better to have someting more generic for measure names, like "perfmon.processor", "perfmon.disk".
And then using tags to add deeper view (like "consumer" in ['user','system','irq', ...] for cpu, or "pattern" in ['read','write'], ...).
And we could also add some other tags like "unit" ['pct','seconds','megabyte_per_sec','io_per_second'...].

That would make querying easier and more human readable.

That is, somewhere, a way to "translate" the perfmon hierarchy.
For that point, it would be also useful to be able to "copy" the "instance" tag values under another tag name (defined at counter level), because sometime we can't just make asumptions on some counter instance, like different disk names, number of cores, network card name, etc. ...

Not sure I am very clear, an example of possible implementation in config files would help you to understand:

```
<CounterGroups>
    <CounterGroup Name="system.cpu" SecondsInterval="5">
        <Counter>
            <CategoryName>Processor</CategoryName>
            <CounterName>% Processor Time</CounterName>
            <InstanceName>*</InstanceName>
            <CounterTag>
                <Name>measure</Name>
                <Value>usage</Value>
            </CounterTag>
            <CounterTag>
                <Name>unit</Name>
                <Value>pct</Value>
            </CounterTag>
            <InstanceTagAlias>
                <Name>core</Name>
            </InstanceTagAlias>
      </Counter>
    </CounterGroup>
</CounterGroups>
etc. ...
```

We would obtain something like:
```
_select * from system.cpu_
time                   measure       unit    core    value   (...)
2016-01-01T01:01:01Z   "usage"       "pct"   "0"     12.564  (...)
2016-01-01T01:01:01Z   "usage"       "pct"   "1"      9.947  (...)
2016-01-01T01:01:01Z   "usage"       "pct"   "2"     28.197  (...)
2016-01-01T01:01:01Z   "usage"       "pct"   "3"     82.073  (...)
2016-01-01T01:01:01Z   "frequency"   "mhz"   "0"     3600    (...)
2016-01-01T01:01:01Z   "frequency"   "mhz"   "1"     3000    (...)
2016-01-01T01:01:01Z   "frequency"   "mhz"   "2"     3000    (...)
2016-01-01T01:01:01Z   "frequency"   "mhz"   "3"     3000    (...)
etc. ...
```

I know it is a big change from the actual approach (which is closer to perfmon view). In fact, in our usage, other system (linuxes, firewalls, apps), will be collected by different tools and we want to get a global approach that make both global analysis and develoment of grafana dashboards easy.

I think this kind of naming convention (all-tagged approach with flexibility at counter level) is more mature and makes sense.

Let me know what you think about that !

**Comment by zeugfr** [comment]
I am sorry if I ask for many enhancements ! As it is for a project I work on, I just asked my boss for a developper for some days to participate on the implementation of these changes, hope it will help to enhance the Influx-Capacitor project without generating you too much work !

**Comment by poxet** [comment]
No problem. You can request anything you like. Since this is a hobby project of mine, I cannot promise when and if it will be implemented. :)

So, if someone else can work on it, that would be great!

All features you have requested so far seems very useful and I am thankful for your input.

**Comment by aderumier** [comment]
Hi,

+1 for this feature.

we are looking too for an windows influxdb collector. I'll try to see if some .net devs at work could help to implement this in coming weeks.

**Comment by poxet** [comment]
Not completley sure if I understood this right, but I have made a try.

From the next release (version 1.0.14) it will be possible to set an attribute named "Alias" on the InstanceName element. Doing so the current field "instance" will still be there, but also have a new field with the name of the alias, and the same values as the "instance".

Something like this...

```
_select * from system.cpu_
time                   instance    core    value   (...)
2016-01-01T01:01:01Z   "0"         "0"     12.564  (...)
2016-01-01T01:01:01Z   "1"         "1"      9.947  (...)
2016-01-01T01:01:01Z   "2"         "2"     28.197  (...)
2016-01-01T01:01:01Z   "3"         "3"     82.073  (...)
2016-01-01T01:01:01Z   "0"         "0"     3600    (...)
2016-01-01T01:01:01Z   "1"         "1"     3000    (...)
2016-01-01T01:01:01Z   "2"         "2"     3000    (...)
2016-01-01T01:01:01Z   "3"         "3"     3000    (...)
etc. ...
```

Did I understand this right?

Still working on the CounterTag...

**Comment by zeugfr** [comment]
Yes, it corresponds to the second part of what I was thinking about and is very usefull !
The other thing I was thinking about is the ability to add custom tags on counter level (not anly application and countergroup).
With these two additions, we will be able to easily create sort of meta metrics (like sys.cpu, sys.disk, mssql.memory ...) with tags to get specific counters or make query with aggregation on several counters.

**Comment by poxet** [comment]
Allright, now you can have CounterTag, CounterGroupTag and GlobalTag. (You can just name them Tag and place them at the level that you want.)

The Alias attribute for instance is also complete.

All is included in version 1.0.14 that was just released,

**Comment by zeugfr** [comment]
Great ! I'll check that asap. Many thanks !
